### DISCUSSION TITLE
Ability to use Zigpy in Zigbee Router mode instead of in Zigbee Coordinator mode?

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#652] [Ideas]
zigpy in "Zigbee Router mode" topic for discussion as Adminiuga and puddly already made some comments around this idea in https://github.com/zigpy/zigpy/pull/589

> > _Or are you thinking of allowing zigpy to configure radios as routers as well as coordinators?_
> 
> _**Yes. Should not assume zigpy will always be a coordinator.**_

Do you think this might also be useful in ZHA if have several Home Assistant instances installations in your house for high-availability?

**Comment by MattWestb** [comment]
EZSP and i think most(all) coordinators can being joining one network as router but i dont see the working case for dng it but perhaps the devs is knowing some like joining one HUE, IKEA or deCONZ network and sniffing, sending and receiving commands. 
Can being one good user case if can doing it nice but i think the LL device "controller bridge" is one better solution for sending / receiving commands autonome in one zigbee network.

One more thing more coordinators with one trustcenter ? 
It should being possible if the zigbee stack can configuration it.

**Comment by MattWestb** [comment]
## IKEA Billy Range Extender firmware
Recommended on ICC-A-1 module but is also working on E1743 remotes "out off the box" but not so good shielded as the module so not good using high output power.

Hardware config:
PB14 TX
PB15 RX
PA0 Button 0 Pairing / Leave button
PA1 Button 1 Not used

Software config:
Standard EZSP Range extender
Added Pull controll cluster.
16 Children (sleeping or not).

Not compleet ZB3 then missing GPProxy that is not in the Silabs profile for the RE.

Flashing new bootloader with SWD.
Then  having SWD connected flash the s37 file or from bootloader.

Botloader switch is IKEA reset / paring switch and also paring / leave in the app.

>(Requires Network Creator Security + Network Creator or Network Steering plugin to function properly in Zigbee 3.0 networks.  Else, Network Find plugin is strongly recommended but not mandatory.)  This code will hook up button 0 to have specific behavior based on the current network state.  The behavior is as follows. If the device is not joined to a network, it will form a Zigbee 3.0 network (via Network Creator plugin) if it is configured as a coordinator or join a Zigbee 3.0 or ZHA legacy network (via Network Steering plugin) if it is configured as a router. If the device is joined to a network then pressing the button will broadcast a ZDO permit join to allow new devices to join.  Holding the button for 5 seconds and releasing will cause the device to leave the network.  Button 1 is not used and a callback is provided to another module wishing to use it.

[ZigbeeBillyRouter.zip](https://github.com/zigpy/zigpy/files/6443565/ZigbeeBillyRouter.zip)

**Comment by Hedda** [comment]
> > _Or are you thinking of allowing zigpy to configure radios as routers as well as coordinators?_
> 
> _**Yes. Should not assume zigpy will always be a coordinator.**_

@Adminiuga Any more thoughts about using zigpy as a Zigbee Router device since the possibility was mentioned in the https://github.com/zigpy/zigpy/pull/589 discussion. If so, are you thinking that it could be useful to have zigpy as a Zigbee Router device joined to a network where zigpy is a Zigbee coordinator as well?

> When zigpy eventually gets support for non-coordinator modes, it'll be good to combine the two and allow zigpy to itself pretend to be the device, connect to the proprietary hub, and just "upgrade" itself to capture the image.

@puddly You suggested https://github.com/zigpy/zigpy/discussions/723 that joining Zigpy as a non-coordinator Zigbee device to an existing Zigbee network where the Zigbee coordinator is a proprietary hubs/gateways/bridges/controllers (like example Samsung SmartThings Hub or Philips Hue Bridge) and let zigpy spoof a third-party device and that way capture OTA firmware image files meant for the original device.

**Reply by MattWestb** [reply]
The last can working or not its depends of the device firmware if accepting OTA from other devices then the IEEE of the trust center (at least its possible setting the Silabs Zigbee stack).

**Comment by Adminiuga** [comment]
Yes, that would be very useful
